src/composable/composable.ts

```diff
@@ -184,6 +184,38 @@ function map<T extends Composable, R>(
   >
 }
 
+/**
+ * Creates a new function that will try to recover from a resulting Failure. When the given function succeeds, its result is returned without changes.
+ * @example
+ * import { cf as C } from 'domain-functions'
+ *
+ * const increment = C.composable(({ id }: { id: number }) => id + 1)
+ * const negativeOnError = C.catchError(increment, (result, originalInput) => (
+ *  originalInput.id * -1
+ * ))
+ */
+function catchError<T extends Fn, R>(
+  fn: Composable<T>,
+  catcher: (
+    err: Omit<Failure, 'success'>,
+    ...originalInput: Parameters<T>
+  ) => R,
+) {
+  return (async (...args: Parameters<T>) => {
+    const res = await fn(...args)
+    if (res.success) return success(res.data)
+    return composable(catcher)(res, ...(args as any))
```

Comment:
Just to make sure you chose it consciously: the first argument to catcher is an actual ErrorResult, not just the errors, right?

Maybe it would be cool to add a test case using the error on the catcher just to feel how the API feels like.

Reply:
That was an oversight, the API is certainly more elegant receiving errors.

Reply:
I have tweaked an existing test case just to properly use the entire API.